ecl/hql/hqlpmap.cpp

```diff
@@ -524,8 +524,15 @@ IHqlExpression * NewProjectMapper2::recursiveExpandSelect(IHqlExpression * expr,
         unsigned match = targets.find(*plain);
         if (match == NotFound)
         {
-            if (ignoreMissingFields)
-                return LINK(expr);
+            /*
+             * Sometimes the implicit project code will remove fields that are used in WILD() expressions.  This creates an inconsistency
+             * in the graph - a reference to a field that doesn't exist in the output of the previous expression.  If we are expanding
+             * a selector reference, and it isn't found in the input's dataset, assume it has been projected out, and remap it to the
+             * parent dataset.  (If this is wrong it will trigger an internal error later on.)
```

Comment:
Are we sure of this assertion - "(If this is wrong it will trigger an internal error later on.)" for all scenarios?